### PR TITLE
Kubernetes v1.7.1

### PR DESCRIPTION
**Pull request by aaronlevy**
- Bumps hyperkube to v1.7.1_coreos.0
- Kube-dns v1.14.4
- Fixing conformance tests to run with v1.7.1
- Vendor bumps:
    - kubernetes v1.70
    - client-go v4.0.0-beta.0

### CODE REVIEW

**Comment by aaronlevy** [comment]
Bumped hyperkube to v1.7.1 but left vendor at v1.7.0 for now (nothing we really need from it).

**Comment by aaronlevy** [comment]
coreosbot run conformance

**Comment by diegs** [comment]
Looks like something really wonky in 1.7.1 when creating the self-hosted components:

`[unable to recognize "/home/core/assets/manifests/etcd-operator.yaml": no matches for extensions/, Kind=Deployment, unable to recognize "/home/core/assets/manifests/kube-apiserver.yaml": no matches for extensions/, Kind=DaemonSet, unable to recognize "/home/core/assets/manifests/kube-calico-role-binding.yaml": no matches for rbac.authorization.k8s.io/, Kind=ClusterRoleBinding, unable to recognize "/home/core/assets/manifests/kube-calico-role.yaml": no matches for rbac.authorization.k8s.io/, Kind=ClusterRole, unable to recognize "/home/core/assets/manifests/kube-calico.yaml": no matches for extensions/, Kind=DaemonSet, unable to recognize "/home/core/assets/manifests/kube-controller-manager-disruption.yaml": no matches for policy/, Kind=PodDisruptionBudget, unable to recognize "/home/core/assets/manifests/kube-controller-manager.yaml": no matches for extensions/, Kind=Deployment, unable to recognize "/home/core/assets/manifests/kube-dns-deployment.yaml": no matches for extensions/, Kind=Deployment, unable to recognize "/home/core/assets/manifests/kube-etcd-network-checkpointer.yaml": no matches for extensions/, Kind=DaemonSet, unable to recognize "/home/core/assets/manifests/kube-flannel.yaml": no matches for extensions/, Kind=DaemonSet, unable to recognize "/home/core/assets/manifests/kube-proxy.yaml": no matches for extensions/, Kind=DaemonSet, unable to recognize "/home/core/assets/manifests/kube-scheduler-disruption.yaml": no matches for policy/, Kind=PodDisruptionBudget, unable to recognize "/home/core/assets/manifests/kube-scheduler.yaml": no matches for extensions/, Kind=Deployment, unable to recognize "/home/core/assets/manifests/kube-system-rbac-role-binding.yaml": no matches for rbac.authorization.k8s.io/, Kind=ClusterRoleBinding, unable to recognize "/home/core/assets/manifests/pod-checkpointer.yaml": no matches for extensions/, Kind=DaemonSet]`

**Comment by aaronlevy** [comment]
Odd. I just ran locally and didn't run into same issue. Gonna re-run tests and see if repeated.

**Comment by aaronlevy** [comment]
coreosbot run e2e

**Comment by aaronlevy** [comment]
coreosbot run conformance

**Comment by aaronlevy** [comment]
I've re-run the conformance test external to this bot (on gce) and all passed. Also the conformance on self-hosted etcd worked as well -- so I'm inclined to ignore that failure (likely random flake).

File: pkg/asset/internal/templates.go [hunk 1]
```diff
@@ -714,6 +714,7 @@ metadata:
   labels:
     k8s-app: kube-dns
     kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
```
Comment (dghubble) [inline pkg/asset/internal/templates.go]: We don't include the addonmanager to my knowledge. If someone adds it, it'd try to manage this I believe.

Comment (aaronlevy) [inline pkg/asset/internal/templates.go]: In bootkube case, I don't think that's a bad thing (if someone wants to run the addonmanager)

Comment (dghubble) [inline pkg/asset/internal/templates.go]: :+1:

**Pull request merged**